### PR TITLE
fix(docs): fix formatting and highlighting issues

### PR DESCRIPTION
**Pull request by eneajaho**
Instead of using `ts` and `html` for code formatting we now use `angular-ts` and `angular-html` which have better support for angular templates + angular inline templates. 

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Code style update (formatting, local variables)
- [x] Documentation content changes

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: adev/src/content/examples/built-in-directives/src/app/app.component.html [hunk 1]
```diff
@@ -190,9 +194,9 @@ <h4 id="ngFor-trackBy">*ngFor trackBy</h4>
 <p>with trackBy and <em>semi-colon</em> separator</p>
 <div class="box">
   <!-- #docregion trackBy -->
-  <div *ngFor="let item of items; trackBy: trackByItems">
-    ({{ item.id }}) {{ item.name }}
-  </div>
+<div *ngFor="let item of items; trackBy: trackByItems">
```
Comment (crisbeto) [inline adev/src/content/examples/built-in-directives/src/app/app.component.html:197]: Maybe I'm missing something, but this looks like it's no longer indented properly. Shouldn't it be one level down from `.box`?

Comment (eneajaho) [reply adev/src/content/examples/built-in-directives/src/app/app.component.html:197]: Before: 
CleanShot 2024-07-26 at 14 50 54@2x

After:
CleanShot 2024-07-26 at 14 51 23@2x


The whole idea is to have the content to show correctly and nicely spaced.

Comment (crisbeto) [reply adev/src/content/examples/built-in-directives/src/app/app.component.html:197]: Interesting, this seems like a bit of a gotcha when authoring new examples. Maybe we should have the docs generation script reduce the indentation instead? It should be doable once #57132 lands.

Comment (JeanMeche) [reply adev/src/content/examples/built-in-directives/src/app/app.component.html:197]: +1 with @crisbeto on this. Idealy input should match the output.

Comment (crisbeto) [reply adev/src/content/examples/built-in-directives/src/app/app.component.html:197]: Or even better: we run the extracted code through Prettier 🤔

Comment (eneajaho) [reply adev/src/content/examples/built-in-directives/src/app/app.component.html:197]: It will still look bad when you extend the code example

Comment (eneajaho) [reply adev/src/content/examples/built-in-directives/src/app/app.component.html]: reverted those changes for now.

**Review by JeanMeche — APPROVED** [review]

**Comment by thePunderWoman** [comment]
This PR was merged into the repository by commit 96ecb8218c680233b6684cfcb39856f8430015c6.

The changes were merged into the following branches: main, 18.1.x

**Pull request closed without merging**